### PR TITLE
Bump cirq version from `0.10.0` to `0.13.0`

### PR DESCRIPTION
**Pull request by vtomole**
Replacing https://github.com/unitaryfund/mitiq/pull/915

Description
-----------

Checklist
-----------

Check off the following once complete (or if not applicable) after opening the PR. The PR will be reviewed once this checklist is complete and all tests are passing.

- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) docstrings for functions.
- [ ] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.

If some items remain, you can mark this a [draft pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

Tips
----

- If the validation check fails:

    1. Run `make check-types` (from the root directory of the repository) and fix any [mypy](https://mypy.readthedocs.io/en/stable/) errors.

    2. Run `make check-style` and fix any [flake8](http://flake8.pycqa.org) errors.

    3. Run `make format` to format your code with the [black](https://black.readthedocs.io/en/stable/index.html) autoformatter.

  For more information, check the [Mitiq style guidelines](https://mitiq.readthedocs.io/en/stable/contributing.html#style-guidelines).
  
- Write "Fixes #XYZ" in the description if this PR fixes Issue #XYZ.

### CODE REVIEW

**Comment by vtomole** [comment]
This change depends on either https://github.com/unitaryfund/mitiq/pull/927 or https://github.com/unitaryfund/mitiq/pull/935.

**Comment by rmlarose** [comment]
Thanks @vtomole. Can you merge master? #992 handled pyquil. Should be good to go after this.

**Comment by nathanshammah** [comment]
Thanks for looking in detail into the breaking issue, @vtomole.

**Comment by vtomole** [comment]
@nathanshammah What do you think is best way to fix this? I think we should raise this issue up to pennylane because `pennylane_from_qasm` https://github.com/unitaryfund/mitiq/blob/8a04147d6430297c2bd8fcdad1834fc69b4efa4c/mitiq/interface/mitiq_pennylane/conversions.py#L84 is giving the incorrect conversion.

**Comment by vtomole** [comment]
Pennylane doesn't support `sx` gate out the box. It compiles it to a ` QubitUnitary` instead

```python

import pennylane

qasm_string = """
// Generated from Cirq v0.13.1

OPENQASM 2.0;
include "qelib1.inc";


// Qubits: [0, 1]
qreg q[2];


rz(0) q[0];
rx(pi*0.1) q[1];
sx q[1];
rz(pi*0.4) q[1];
s q[1];
cx q[0],q[1];
"""

qfunc = pennylane.from_qasm(qasm_string)

with pennylane.tape.QuantumTape() as tape:
        qfunc(wires=pennylane.wires.Wires(range(2)))

print(tape.operations)
# prints
# [RZ(0.0, wires=[0]), 
# RX(0.3141592653589793, wires=[1]), 
# QubitUnitary(array([[0.5+0.5j, 0.5-0.5j],
#        [0.5-0.5j, 0.5+0.5j]]), wires=[1]), 
#        RZ(1.2566370614359172, wires=[1]), 
#        S(wires=[1]), 
#        CNOT(wires=[0, 1])]
```

**Comment by vtomole** [comment]
Not a solution; but to get this PR in we could remove the `cirq.rx(0.5 * np.pi).on(b) operation,  # Clifford` out of this test.

**Comment by vtomole** [comment]
The fix: https://github.com/PennyLaneAI/pennylane-qiskit/pull/158

**Comment by andreamari** [comment]
> The fix: [PennyLaneAI/pennylane-qiskit#158](https://github.com/PennyLaneAI/pennylane-qiskit/pull/158)

Thanks for asking for this fix in PL, @vtomole! Given this fact, I think we just need to wait from the Mitiq side.

**Comment by nathanshammah** [comment]
Great suggestion, @vtomole!

**Comment by vtomole** [comment]
> Given this fact, I think we just need to wait.

> Great suggestion

I'm getting conflicting information. Should we wait for that Pennylane PR to get merged and for Pennylane to cut a new release before we merge this or should we modify the test here so it passes for us?

**Comment by nathanshammah** [comment]
@vtomole I referred to the idea of upstreaming this to Pennylane and the issue you opened on Pennylane. Let's wait for that release. We can also discuss at the Mitiq meeting if you want to join, at 12pm ET on http://discord.unitary.fund/

**Comment by vtomole** [comment]
Okay, thanks for clearing that up!

**Comment by vtomole** [comment]
@rmlarose @nathanshammah @rmlarose Ready for re-review :slightly_smiling_face:

**Comment by nathanshammah** [comment]
Probably this will have the same issue of #1048(?), the `pennylane` update. #1049 updates `pennylane-qiskit`.

**Comment by nathanshammah** [comment]
@dependabot rebase

**Comment by andreamari** [comment]
@crazy4pi314 @nathanshammah  
In previous commits, only greetings started in CI due to GitHub security rules for external contributions. However there was a an way to manually trigger all tests. I don't see this option anymore. Do you see this option or know how to run all tests of the last commit?

**Comment by crazy4pi314** [comment]
> However there was a way to manually trigger all tests. I don't see this option anymore. Do you see this option or know how to run all tests of the last commit?
Yeah I agree we used to see this: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks#approving-workflow-runs-on-a-pull-request-from-a-public-fork

Was it running tests b4 asking for the rebase?

**Comment by vtomole** [comment]
The docs check is failing with a 

> Command killed due to excessive memory consumption

message.

**Comment by crazy4pi314** [comment]
> The docs check is failing with a
> 
> > Command killed due to excessive memory consumption
> 
> message.

This can randomly happen when one of the examples has a call to `scipy.fit` that does not converge, often re-running (which I just kicked off) should allow it to proceed if that was the issue.

**Comment by vtomole** [comment]
@nathanshammah @andreamari Ready for re-review :)

File: mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py
Comment (andreamari) [inline mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py:125]:
```suggestion
        qml.CRX(0.4, wires=(0, 1)),
```
To avoid an operation equivalent to the identity.

File: mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py
Comment (andreamari) [inline mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py:145]: Nice!

File: mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py
Comment (andreamari) [inline mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py:149]: I guess you had to remove this for some tricky reason.
If not possible to re-introduce:
- Can we restore the random rotation parameters only?
- Can we restore the random wires only?

If not, it's fine. This is not a blocker for this PR.

Comment (vtomole) [reply mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py:149]: The reason for this change is because `gate.num_params` is now dynamic. Since this test is copied from pennylane, the following PR recommends re-writing the test this way. See https://github.com/PennyLaneAI/pennylane/pull/1898/files#r750946650.

File: mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py
Comment (vtomole) [reply mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py]: Done.

File: mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py
Comment (andreamari) [reply mitiq/interface/mitiq_pennylane/tests/test_conversions_pennylane.py:149]: Ah, I see, very nice finding!

**Review by andreamari — APPROVED** [review]
Many thanks @vtomole! LGTM
(running all the tests now, we can merge once they pass)

**Comment by nathanshammah** [comment]
Thanks @vtomole! LGTM. I opened #1112 to check that RTD does not fail on all PRs. I will restart the checks to see if now they pass. It was killed for eccessive memory consumption.

**Comment by andreamari** [comment]
Hi @astrojuanlu, do you know why RDT is failing for this PR (and only for this PR)? 

My guess, please correct me if I am wrong, is this: due to a slightly longer pip install process, the docs build of this PR exceeds the 15 minutes limit that is discussed at this page (https://docs.readthedocs.io/en/stable/builds.html).
In the same page it is also written that: 

_We can increase build limits on a per-project basis. Send an email to support@readthedocs.org providing a good reason why your documentation needs more resources._

Do you think we can get some extra build time? This would help us unblock urgent PRs until we find a more scalable solution for the docs.

**Comment by astrojuanlu** [comment]
Hi @andreamari ! It looks like the other builds are on the edge of the 15 minute limit. Looks like most of the build time is spent executing notebooks, which makes a ton of sense. Please send an email to support@readthedocs.org and the support team will take care of it :)

**Comment by nathanshammah** [comment]
Thanks @astrojuanlu for the info and @andreamari for investigating. 


If we switch in the future to Jupyter book, it may be simpler to cache notebooks (at least for some CI checks).

**Comment by andreamari** [comment]
> Hi @andreamari ! It looks like the other builds are on the edge of the 15 minute limit. Looks like most of the build time is spent executing notebooks, which makes a ton of sense. Please send an email to [support@readthedocs.org](mailto:support@readthedocs.org) and the support team will take care of it :)

I just sent and email to [support@readthedocs.org](mailto:support@readthedocs.org, thanks for the advice!

**Review by nathanshammah — APPROVED** [review]
Thank you @vtomole, all tests now pass.

**Comment by andreamari** [comment]
And also thanks to @astrojuanlu (RTD is working now)!

**Comment by vtomole** [comment]
@nathanshammah @andreamari Thanks for your patience!

**Comment by andreamari** [comment]
@allcontributors please add @vtomole for tests and code

**Pull request merged**
